src/Sparrow.Server/DynamicNativeLibraryResolver.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Sparrow.Server
+{
+    public static class DynamicNativeLibraryResolver
+    {
+        private static HashSet<string> _registered = new HashSet<string>();
+        public static void Register(string lib)
+        {
+            lock (typeof(DynamicNativeLibraryResolver))
+            {
+                var copy = new HashSet<string>(_registered) {lib};
+                _registered = copy;
+                if (copy.Count != 1)
+                    return;
+                NativeLibrary.SetDllImportResolver(typeof(Sodium).Assembly, Resolver);
+            }
+        }
+
+        private static IntPtr Resolver(string libraryName, Assembly assembly, DllImportSearchPath? dllImportSearchPath)
+        {
+            if(_registered.Contains(libraryName) == false)
+                return IntPtr.Zero;
+
+            string suffix;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                if (RuntimeInformation.ProcessArchitecture != Architecture.Arm &&
+                    RuntimeInformation.ProcessArchitecture != Architecture.Arm64)
+                {
+                    suffix = Environment.Is64BitProcess ? ".linux.x64.so" : ".linux.x86.so";
+                }
+                else
+                {
+                    suffix = Environment.Is64BitProcess ? ".arm.64.so" : ".arm.32.so";
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                suffix = Environment.Is64BitProcess ? ".mac.x64.dylib" : ".mac.x86.dylib";
+                // in mac we are not : `toFilename += ".so";` as DllImport doesn't assume .so nor .dylib by default
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                suffix = Environment.Is64BitProcess ? ".win.x64.dll" : ".win.x86.dll";
```

Comment:
Missing handling of `win7` here, which is now out of support.  Not sure if we want to remove the `win7` support or keep it. Thoughts?

Reply:
From one side yes, from the other I have a feeling that it might still be popular in come circles. There is extended support option available? Or Windows Server equivalent that has still support?

Reply:
As long as .NET core supports it, it isn't hard to support it too